pontoon/base/management/commands/update_auth_providers.py

```diff
@@ -0,0 +1,75 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file has been copied from http://github.com/mozilla/testpilot
+# and has been modified in order to use Fxa provider from the allauth package.
+from __future__ import absolute_import
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.contrib.sites.models import Site
+from allauth.socialaccount.models import SocialApp
+from allauth.socialaccount.providers.fxa.provider import FirefoxAccountsProvider
+from allauth.socialaccount.providers.github.provider import GithubProvider
+
+
+FXA_PROVIDER_ID = FirefoxAccountsProvider.id
+GITHUB_PROVIDER_ID = GithubProvider.id
+
+
+class Command(BaseCommand):
+    help = ('Ensures an allauth application exists and has credentials that match settings')
+
+    def handle(self, *args, **options):
+        def update_provider(data):
+            try:
+                # Update the existing provider with current settings.
+                app = SocialApp.objects.get(provider=data['provider'])
+                self.stdout.write("Updating existing authentication provider (pk=%s)" % app.pk)
+                for k, v in data.items():
+                    setattr(app, k, v)
+                app.save()
+            except ObjectDoesNotExist:
+                # Create the provider if necessary.
+                app = SocialApp(**data)
+                app.save()
+                self.stdout.write("Created new authentication provider (pk=%s)" % app.pk)
+
+            # Ensure the provider applies to the current default site.
+            sites_count = app.sites.count()
+            if sites_count == 0:
+                default_site = Site.objects.get(pk=settings.SITE_ID)
+                app.sites.add(default_site)
+
+        if 'FXA':
```

Comment:
What I meant with my pseudocode example was to use the invert of the if clause below, something like:
```python
if settings.FXA_CLIENT_ID is not None or settings.FXA_SECRET_KEY is not None:
  # do_something()
```

Reply:
OK.  Would the line "self.stdout.write("FXA_* settings unavailable; skipping provider config.")" not be needed then, or would it go into an `else`

Reply:
It would go into `else`. We'd also need to make sure at least one provider is available before running `# Ensure the provider applies to the`.

Reply:
Hmm...so if the FXA ID and KEY are `None` then the `else` will trigger the write line even if the GitHub ID and KEY are `not None`? Is that correct??

If so, should it be an `if/elif/else` block with the `else` writing a general "settings unavailable; skipping provider config." message?

Reply:
If it makes things easier, you can also log on success:
```python
if settings.FXA_CLIENT_ID is not None or settings.FXA_SECRET_KEY is not None:
  # do_something()
  # log_something()
if settings.GITHUB...:
  # do_something()
  # log_something()
```

Reply:
I went ahead and made the changes that I thought would maintain the originally intended function to bail if no settings found.  I just pushed those changes, so just let me know if anything needs revised.  Thanks.